watch/inotify.go

```diff
@@ -96,12 +96,14 @@ func (fw *InotifyFileWatcher) ChangeEvents(t *tomb.Tomb, pos int64) (*FileChange
 			}
 
 			switch {
-			//With an open fd, unlink(fd) - inotify returns IN_ATTRIB (==fsnotify.Chmod)
 			case evt.Op&fsnotify.Chmod == fsnotify.Chmod:
-				if _, err := os.Stat(fw.Filename); err != nil {
-					if !os.IsNotExist(err) {
-						return
-					}
+				_, err = os.Stat(fw.Filename)
+				if err == nil {
+					continue
+				}
+				// With an open fd, unlink(fd) - inotify returns IN_ATTRIB (==fsnotify.Chmod)
+				if !os.IsNotExist(err) {
```

Comment:
Good point. The original source has code that indicates that the author does not know how to handle this. I an going to deprecate this PR in favor of hpcloud/tail#112. That PR re-uses the original code's currently awful way of reporting such errors, but at least it's consistent.